src/sentry/notifications/notificationcontroller.py

```diff
@@ -506,3 +506,50 @@ def user_has_any_provider_settings(self, provider: ExternalProviderEnum | None =
                 return True
 
         return False
+
+    def get_notification_value_for_recipient_and_type(
+        self, recipient: Recipient, type: NotificationSettingEnum
+    ) -> NotificationSettingsOptionEnum:
+        """
+        Returns the notification setting value for the given recipient and type.
+
+        Args:
+            recipient: The recipient of the notification settings (user or team).
+            type: The notification type to filter providers and recipients by.
+        """
+        if self.type and type != self.type:
+            raise Exception("Type mismatch: the provided type differs from the controller type")
+
+        setting_options = self._get_layered_setting_options(type=type.value)
+        for _, setting in setting_options[recipient].items():
```

Comment:
@snigdhas I know I wrote this code, but is it right to ignore the scope?

Reply:
yep, we don't need to use the scope unless we plan on iterating over settings for multiple projects